packages/core/src/utils/BlockchainApiUtil.ts

```diff
@@ -0,0 +1,27 @@
+import { ConfigCtrl } from '../controllers/ConfigCtrl'
+
+// -- Constants -------------------------------------------------------
+export const NAMESPACE = 'eip155'
+const BLOCKCHAIN_API = 'https://rpc.walletconnect.com'
+
+// -- Utility -------------------------------------------------------
+export const BlockchainApiUtil = {
+  async getIdentity(
+    address: string,
+    chainId: number
+  ): Promise<{ name: string; avatar?: string } | null> {
+    const { projectId } = ConfigCtrl.state
+    const chain_id = `${NAMESPACE}:${chainId}`
+    const endpoint = `${BLOCKCHAIN_API}/v1/identity/${address}?chainId=${chain_id}&projectId=${projectId}`
+
+    const response = await fetch(endpoint)
+    if (response.status === 404) {
```

Comment:
Actually, since 404 errors log to the console, I think this should be part of the JSON response not a status code. Also, it enables differentiating the route not being found vs profile not being found.

![image](https://github.com/WalletConnect/web3modal/assets/1979423/ea8fde3b-0201-4a06-9bb2-08e3c5958ebe)

Reply:
Fixed by https://github.com/WalletConnect/rpc-proxy/pull/222